pkg/apis/serving/v1/revision_types.go

```diff
@@ -129,13 +129,31 @@ type RevisionStatus struct {
 	// +optional
 	LogURL string `json:"logUrl,omitempty"`
 
-	// ImageDigest holds the resolved digest for the image specified
+	// DeprecatedImageDigest holds the resolved digest for the image specified
 	// within .Spec.Container.Image. The digest is resolved during the creation
 	// of Revision. This field holds the digest value regardless of whether
 	// a tag or digest was originally specified in the Container object. It
 	// may be empty if the image comes from a registry listed to skip resolution.
+	// If multiple containers specified then DeprecatedImageDigest holds the digest
+	// for serving container.
+	// DEPRECATED Use ImageDigests instead.
+	// TODO(savitaashture) Remove deprecatedImageDigest.
+	// ref https://kubernetes.io/docs/reference/using-api/deprecation-policy for deprecation.
 	// +optional
-	ImageDigest string `json:"imageDigest,omitempty"`
+	DeprecatedImageDigest string `json:"imageDigest,omitempty"`
+
+	// ImageDigests is a mapping of images present in .Spec.Container[*].Image
+	// to their respective digests. The digests are resolved during the creation
+	// of Revision. ImageDigests holds the digests for both serving and non serving container.
+	// ref: http://bit.ly/image-digests
+	// +optional
+	ImageDigests []ImageDigests `json:"imageDigests,omitempty"`
+}
+
+// ImageDigests holds the information of container name and digest value
+type ImageDigests struct {
+	ContainerName string `json:"containerName,omitempty"`
+	ImageDigest   string `json:"imageDigest,omitempty"`
```

Comment:
@dprotaso 
When we do 
`kubectl get pods multicontainer-v1-deployment-56b6cdcdb-dpbkr -o yaml`
in the `containerStatuses` we have the imageID which is similar to **rev** imageDigest

But the structure which declared in this PR is to show in the imageDigests in **rev status** and also the same structure data will help to create a **image cache**

Now when you say extension to container status do you mean something we need to integrate with that ?

Reply:
> Now when you say extension to container status do you mean something we need to integrate with that ?

To clarify, I'm not suggesting we embed the PodSpec's ContainerStatus in our Revision Status. I think a new struct that we maintain (that'll currently contain Name and ImageDigest).

I'm tempted to do it this way since users may want to surface additional properties of the container up to the revision status. This new struct is a way to consolidate the properties associated with a container in one place.

Reply:
Okay so my understanding here is
you are proposing to use struct like below
```
type ContainerStatus struct {
	Name string `json:"name,omitempty"`
	ImageDigest   string `json:"imageDigest,omitempty"`
}
```
instead of 
```
type ImageDigests struct {
	ContainerName string `json:"containerName,omitempty"`
	ImageDigest   string `json:"imageDigest,omitempty"`
}
```
So that if we want we can add required properties of container to `ContainerStatus` of `rev` status

@dprotaso  then i feel we also need to update these changes to design doc [here](https://docs.google.com/document/d/1XjIRnOGaq9UGllkZgYXQHuTQmhbECNAOk6TT6RNfJMw/edit?ts=5e25d093#)

Reply:
@dprotaso Addressed all the comments
Please take a look 
Thank you